Update all dependencies to their latest versions

rfratto opened the pull request.

rfratto committed:
Update github.com/go-kit/log to latest

Move away from github.com/go-kit/kit to the log module, which is
smaller.

rfratto committed:
update github.com/apache/pulsar-client-go dependency

rfratto committed:
update github.com/linkedin/goavro to latest

rfratto committed:
Update github.com/prometheus/common to latest

rfratto committed:
Update github.com/sirupsen/logrus to latest

rfratto committed:
update github.com/stretchr/testify to latest

rfratto commented:
It looks like I need to revisit the CI and potentially update that too. All the tests were passing locally for me (except the integration tests, which I need to look into).

tpaschalis commented:
[Similar](https://github.com/dapr/components-contrib/issues/1639) CI failure due to the linter in other repo. I think it might pass when the new linter version is merged.

rfratto committed:
Merge branch 'master' into update-deps

rfratto committed:
handle ack errors

The Ack method can now return an error which needs to be handled. The
error is ignored when acknowledging a bad sample.

rfratto committed:
fixup! update github.com/apache/pulsar-client-go dependency

rfratto committed:
`make update-readme`

tpaschalis reviewed (approved):
LGTM
